src/main/java/it/gov/pagopa/pu/organization/model/DistinctCollectionReasonDTO.java

```diff
@@ -0,0 +1,25 @@
+package it.gov.pagopa.pu.organization.model;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+@Entity
+@Table(name = "taxonomy")
+public class DistinctCollectionReasonDTO {
+
+  @Id
+  private String organizationType;
+  private String organizationTypeDescription;
+  private String macroAreaCode;
+  private String macroAreaName;
+  private String macroAreaDescription;
+  private String serviceTypeCode;
+  private String serviceType;
+  private String serviceTypeDescription;
+  private String collectionReason;
```

Comment:
the Id should be the distinct columns.
Can you check if annotating the class with @IdClass(TaxonomyCollectionReasonDTO.class)
and setting the following Id if it works?

If works, we should do this in all DTO, except OrganizationTypeDTO
```suggestion
  @Id
  private String organizationType;
  private String organizationTypeDescription;
  @Id
  private String macroAreaCode;
  private String macroAreaName;
  private String macroAreaDescription;
  @Id
  private String serviceTypeCode;
  private String serviceType;
  private String serviceTypeDescription;
  @Id
  private String collectionReason;
```